blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClient.scala

```diff
@@ -12,33 +12,31 @@ import scalaz.{-\/, \/-}
 object BlazeClient {
   def apply(manager: ConnectionManager, idleTimeout: Duration, requestTimeout: Duration): Client = {
     Client(Service.lift { req =>
-      def tryClient(client: BlazeClientStage, freshClient: Boolean, flushPrelude: Boolean): Task[DisposableResponse] = {
+      val key = RequestKey.fromRequest(req)
+      def tryClient(client: BlazeClientStage, flushPrelude: Boolean): Task[DisposableResponse] = {
         // Add the timeout stage to the pipeline
         val ts = new ClientTimeoutStage(idleTimeout, requestTimeout, bits.ClientTickWheel)
         client.spliceBefore(ts)
         ts.initialize()
 
         client.runRequest(req, flushPrelude).attempt.flatMap {
-          case \/-(r)    =>
+          case \/-(r)  =>
             val dispose = Task.delay {
-              if (!client.isClosed()) {
-                ts.removeStage
```

Comment:
The state is still `Running`.  My call is still not touching the body, so I never get to any of the cleanup here.  We still lose if we don't run the body.

https://github.com/http4s/http4s/blob/v0.11.3/blaze-client/src/main/scala/org/http4s/client/blaze/Http1ClientStage.scala#L192-L209